src/AppBundle/Repository/PostRepository.php

```diff
@@ -38,4 +38,24 @@ public function findLatest($limit = Post::NUM_ITEMS)
             ->getResult()
         ;
     }
+
+    public function findByTerms(array $terms, $limit = Post::NUM_ITEMS)
+    {
+        $queryBuilder = $this->createQueryBuilder('p');
+
+        $i = 0;
+        foreach ($terms as $term) {
```

Comment:
We could use unique index on traversing `$terms` like:
`foreach ($terms as $i => $term) {`